src/WCSimWCAddDarkNoise.cc

```diff
@@ -0,0 +1,254 @@
+#include "WCSimWCAddDarkNoise.hh"
+#include "WCSimWCPMT.hh"
+#include "WCSimWCDigi.hh"
+#include "WCSimWCHit.hh"
+#include "G4EventManager.hh"
+#include "G4Event.hh"
+#include "G4SDManager.hh"
+#include "G4DigiManager.hh"
+#include "G4ios.hh"
+#include "G4RotationMatrix.hh"
+#include "G4ThreeVector.hh"
+#include "G4LogicalVolumeStore.hh"
+
+#include "WCSimDetectorConstruction.hh"
+#include "WCSimPmtInfo.hh"
+#include "WCSimDarkRateMessenger.hh"
+
+#include <vector>
+#include <utility>
+// for memset
+#include <cstring>
+#include <iostream>
+
+#ifndef WCSIMWCADDDARKNOISE_VERBOSE
+//#define WCSIMWCADDDARKNOISE_VERBOSE
+#endif
+
+WCSimWCAddDarkNoise::WCSimWCAddDarkNoise(G4String name,
+                                   WCSimDetectorConstruction* myDetector)
+  :G4VDigitizerModule(name)
+{
+  this->myDetector = myDetector;
+  DigiHitMap.clear();
```

Comment:
Hmm you're right. The purpose of the 'Default' in a G4UIcmdWithA* is a mystery to me then...
I've just committed a fix where the defaults in WCSimWCAddDarkNoise are set when the constructor of WCSimDarkRateMessenger is called. I've done it this way (rather than being explicit in the WCSimWCAddDarkNoise constructor) so as to not have to change defaults in 2 places in the code (which is of course likely to get out of sync).